README.md

```diff
@@ -51,14 +51,25 @@ There are tests for `gcc` and `clang` in `tests/gcc` and `tests/clang` respectiv
 
 ## Build input hook
 
-It is possible to use mini compile commands to generate a `compile_commands.json` while building a derivation. This involves using a wrapped standard standard environment and adding a hook to the `buildInputs`:
-```
-with import <nixpkgs> {};
-let mcc-env = (callPackage <this_repo> {}).wrap stdenv;
-    mcc-hook = (callPackage <this_repo> {}).hook;
-in (hello.override { stdenv = mcc-env; }).overrideAttrs
+It is possible to use mini compile commands to generate a `compile_commands.json` while building a derivation. This works by building directly in a subdirectory of `$out` or some other `$custom-output` (so that `compile_commands.json` contains valid paths).
```

Comment:
This seems incredibly suspicious to me. We don't want to interfere with the natural build process in any way, other than to pull information out of the compiler wrapper. For example, what if the standard build directory is on a fast, local file system and the nix store is on some remote, slow file system? This will almost surely unearth a lot of build time concurrency bugs

Reply:
when building nix uses a mount on /tmp, so this is still running on the local file system. after the build, if successful, the result is copied to the actual store

Reply:
Ah OK, this makes sense.